### ISSUE TITLE
Use __attribute((__malloc__)) not __attribute((malloc)) in macros.h

### ISSUE THREAD

**Issue by marka63** [openssl/openssl#20776]
`#   define OSSL_CRYPTO_ALLOC __attribute__((malloc))` in include/openssl/macros.h interacts badly with anything that defines malloc like cmocka's cmocka.h .  GCC recommends the use of `__attribute__((__malloc__))` in header files to  prevent this issue https://gcc.gnu.org/onlinedocs/gcc-4.7.2/gcc/Function-Attributes.html.

`You may also specify attributes with '__' preceding and following each keyword. This allows you to use them in header files without being concerned about a possible macro of the same name. For example, you may use __noreturn__ instead of noreturn.
`

```
% grep malloc /opt/local/include/cmocka.h 
 *      int *answer = malloc(sizeof(int));
 *      int *answer = malloc(sizeof(int));
 * tested by cmocka should replace calls to malloc(), calloc() and free() to
 * test_malloc(), test_calloc() and test_free() respectively. Each time a block
 * @brief Test function overriding malloc.
 * extern void* _test_malloc(const size_t size, const char* file, const int line);
 * #define malloc(size) _test_malloc(size, __FILE__, __LINE__)
 *     int * const temporary = (int*)malloc(sizeof(int));
 * @see malloc(3)
void *test_malloc(size_t size);
#define test_malloc(size) _test_malloc(size, __FILE__, __LINE__)
/* Redirect malloc, calloc and free to the unit test allocators. */
#define malloc test_malloc
void* _test_malloc(const size_t size, const char* file, const int line);
% 
```

**Comment by t8m** [comment]
This is regression in 3.1.

**Comment by Lipovlan** [comment]
Hi @Anis-cpu-13,

I'm no maintainer, but submitting [OpenSSL's CLA](https://www.openssl.org/policies/cla.html) is a good start for contributing non-trivial stuff.

Once you have yours submitted, go ahead and work on any issue you think you are fit for. If it is not anything big (which stuff marked with "good first issue" is not), you can simply do the work without prior consultation and then open a pull-request.

If the maintainers do have problems with your commits, they will just tell you and you can either argue for your cause or do the necessary changes to comply.

Don't forget to mention the issue your pull-request solves in the description so it can be closed :)

**Comment by paulidale** [comment]
This would likely fall under the trivial clause.  Submit a pull request and we'll decide.  Or better start with your ICLA...

**Comment by Anis-cpu-13** [comment]
Thanks a bunch for the useful tips on getting started with contributing to the OpenSSL project! I really appreciate you taking the time to share this info. I'll be sure to follow your suggestions, like submitting the CLA and mentioning the issue in my pull-request.
Can't wait to dive in and learn from this experience! Thanks again for your help, mate

**Comment by Anis-cpu-13** [comment]
Fix in #20869
